commit 145a24fffb1ed2458d887b22a5202f4f648051e0
Author: Boyuan Yang <073plan@gmail.com>
Date:   Thu Jan 25 22:13:11 2018 +0800

    src: engine: position.cpp: cherry-pick upstream bugfix
    
    See also
    github.com/xqbase/eleeye/commit/f460861b1bff7a353d36f09f3610d4dd042d003e

diff --git a/src/engine/position.cpp b/src/engine/position.cpp
index 0732a85..06563d6 100644
--- a/src/engine/position.cpp
+++ b/src/engine/position.cpp
@@ -467,12 +467,9 @@ void PositionStruct::FromFen(const char *szFen) {
         break;
       }
     } else if (*lpFen >= '1' && *lpFen <= '9') {
-      for (k = 0; k < (*lpFen - '0'); k ++) {
-        if (j >= FILE_RIGHT) {
-          break;
-        }
-        j ++;
-      }
+/* https://github.com/xqbase/eleeye/commit/
+ * f460861b1bff7a353d36f09f3610d4dd042d003e */
+      j += (*lpFen - '0');
     } else if (*lpFen >= 'A' && *lpFen <= 'Z') {
       if (j <= FILE_RIGHT) {
         k = FenPiece(*lpFen);
